cpp/src/DataStorm/SessionI.h

```diff
@@ -9,13 +9,12 @@
 #include "Ice/Ice.h"
 #include "NodeI.h"
 
-#if defined(_MSC_VER)
-#    pragma warning(push)
-#    pragma warning(disable : 4456) // ... : declaration of 'identifier' hides previous local declaration
-#    pragma warning(disable : 4458) // ... : declaration of 'identifier' hides class member
-#elif defined(__clang__)
+#if defined(__clang__)
 #    pragma clang diagnostic push
-#    pragma clang diagnostic ignored "-Wshadow"
```

Comment:
We should never disable plain -Wshadow, especially not in a header file. I fixed the shadowing in this PR.